client/clip_client/client.py

```diff
@@ -110,14 +111,21 @@ def encode(self, content, **kwargs):
             )
 
         self._prepare_streaming(
-            not kwargs.get('show_progress'),
+            not kwargs.pop('show_progress'),
             total=len(content) if hasattr(content, '__len__') else None,
         )
         results = DocumentArray()
+
+        parameters = kwargs.pop('parameters', {})
```

Comment:
alright, `pop` operator would raise an exception when the given `key` does not exist.